cristin-project/src/main/java/no/unit/nva/cristin/projects/update/UpdateProjectResourceAccessCheck.java

```diff
@@ -24,24 +24,25 @@ public class UpdateProjectResourceAccessCheck extends ProjectResourceAccessCheck
 
     private transient boolean verified;
 
+    @SuppressWarnings({"PMD.DataflowAnomalyAnalysis"})
     @Override
     public void verifyAccess(NvaProject resource, Map<String, String> params) {
         var userIdentifier = params.get(USER_IDENTIFIER);
 
-        var resourceCreator = getResourceCreator(resource);
-        var resourceManager = getResourceManager(resource);
-
-        if (resourceCreator.isPresent() && hasMatch(resourceCreator.get(), userIdentifier)) {
-            verified = true;
+        getResourceCreator(resource)
+            .ifPresentOrElse(creator -> verified = hasMatch(creator, userIdentifier),
+                             () -> verified = false);
+        if (verified) {
             return;
-        } else if (resourceManager.isPresent() && hasMatch(resourceManager.get(), userIdentifier)) {
-            verified = true;
+        }
+
+        getResourceManager(resource)
+            .ifPresent(manager -> verified = hasMatch(manager, userIdentifier));
+        if (verified) {
             return;
         }
 
         logger.info(NOT_ALLOWED_TO_UPDATE_THIS_RESOURCE, userIdentifier, resource.getId());
-
-        verified = false;
```

Comment:
im a little superstitious and think we should set this to false at the end just in case. Or else we have to rely on always creating a new instance of the class every time we want to verify

Reply:
```
.ifPresentOrElse(creator -> verified = hasMatch(creator, userIdentifier),
                 () -> verified = false);
```
verified will always have a value, and I am pretty sure boolean null evaluate to false in java

Reply:
Well that is if you always create a new instance of the class when you do validation. If you reuse an old instance it can use an older calculation and then it might be already set to true